### PR TITLE
update the error handler to receive all exceptions

### PR DESCRIPTION
**Pull request by Poolitzer**

### CODE REVIEW

**Comment by Poolitzer** [comment]
(implements #1173)

File: telegram/ext/dispatcher.py [hunk 1]
```diff
@@ -337,37 +337,42 @@ def process_update(self, update):
...
 
-            # Dispatch any error.
-            except TelegramError as te:
-                self.logger.warning('A TelegramError was raised while processing the Update')
-
+            # Dispatch any error - errors should not stop the thread.
+            except Exception as e:
                 try:
-                    self.dispatch_error(update, te)
-                except DispatcherHandlerStop:
```
Comment (Eldinnie) [inline telegram/ext/dispatcher.py]: The Error handler can still raise `DispatcherHandlerStop`.
This should still be in this except clause.

File: telegram/ext/dispatcher.py [hunk 2]
```diff
@@ -439,11 +444,15 @@ def update_persistence(self):
                 self.persistence.update_user_data(user_id, self.user_data[user_id])
 
     def add_error_handler(self, callback):
-        """Registers an error handler in the Dispatcher.
+        """Registers an error handler in the Dispatcher. This handler will receive every error
+        which happens in your bot.
```
Comment (Eldinnie) [inline telegram/ext/dispatcher.py]: which -> that

File: telegram/ext/dispatcher.py [hunk 3]
```diff
@@ -469,7 +478,8 @@ def dispatch_error(self, update, error):
 
         Args:
             update (:obj:`str` | :class:`telegram.Update` | None): The update that caused the error
-            error (:class:`telegram.TelegramError`): The Telegram error that was raised.
+            error (:class:`telegram.TelegramError` | :Type:`Exception`): The (Telegram) error
+            that was raised.
```
Comment (Eldinnie) [inline telegram/ext/dispatcher.py]: just :obj:`Exception` will do. No need for both

File: telegram/ext/jobqueue.py [hunk 4]
```diff
@@ -172,6 +172,11 @@ def run_repeating(self, callback, interval, first=None, context=None, name=None)
             :class:`telegram.ext.Job`: The new ``Job`` instance that has been added to the job
             queue.
 
+        Notes:
```
Comment (Eldinnie) [inline telegram/ext/jobqueue.py]: The notes in jobqueue are unrelated so please remove them from this PR

File: telegram/ext/updater.py [hunk 5]
```diff
@@ -383,7 +383,7 @@ def _start_webhook(self, listen, port, url_path, cert, key, bootstrap_retries, c
             ssl_ctx = None
 
         # Create and start server
-        self.httpd = WebhookServer(port, app, ssl_ctx)
+        self.httpd = WebhookServer(listen, port, app, ssl_ctx)
```
Comment (Eldinnie) [inline telegram/ext/updater.py]: Unrelated change, please remove from this PR

File: telegram/utils/webhookhandler.py [hunk 6]
```diff
@@ -34,8 +34,9 @@
 
 class WebhookServer(object):
 
-    def __init__(self, port, webhook_app, ssl_ctx):
+    def __init__(self, listen, port, webhook_app, ssl_ctx):
```
Comment (Eldinnie) [inline telegram/utils/webhookhandler.py]: Unrelated changes, please remove from this PR.

File: telegram/ext/jobqueue.py (hunk 4 above)
Comment (Poolitzer) [reply telegram/ext/jobqueue.py]: done, sorry

File: telegram/ext/updater.py (hunk 5 above)
Comment (Poolitzer) [reply telegram/ext/updater.py]: done, sorry

File: telegram/utils/webhookhandler.py (hunk 6 above)
Comment (Poolitzer) [reply telegram/utils/webhookhandler.py]: done, sorry

File: telegram/ext/dispatcher.py (hunk 1 above)
Comment (Poolitzer) [reply telegram/ext/dispatcher.py]: it is again

File: telegram/ext/dispatcher.py (hunk 3 above)
Comment (Poolitzer) [reply telegram/ext/dispatcher.py]: good point

File: telegram/ext/dispatcher.py (hunk 2 above)
Comment (Poolitzer) [reply telegram/ext/dispatcher.py]: that it is
